code/modules/reagents/chemistry_reagents/drink.dm

```diff
@@ -211,21 +211,23 @@
 
 /datum/reagent/consumable/drink/coffee/on_mob_life(mob/living/L, metabolism)
 	L.jitter(2)
+	L.reagent_move_delay_modifier -= min(0.5, volume * 0.0033) //150u for FULL speedboost, .3 is the max "safe" achievable.
```

Comment:
The design for coffee is both punishing and strong.
Make the benefit (the increased speed) be added on `on_mob_add` and removed on `on_mob_delete`, but give a penalty on delete.
Also give the maluses on `on_overdose_start`, but make them temporary issues, such as worse aim, blurred vision or even a slowdown.
That way you need to keep using small doses to keep the benefits but you'll suffer either if you stop drinking or overdose.
You might want to wait for spooky to port the /tg/ status effects, however.

Reply:
I'll go over changing the speed boost to on_mob_add/delete once I see how it plays as-is. I'd like to keep the speed boost on THIS one more relevant to dosage. I think changing it to a flat speed boost per level (.1 per in your system/OD'd/crit OD'd) is what I wa… OOH!

I'll change it so you get 
-.1 boost if you have less than 5u OR you're OD'd.
-.2 boost if OD'd within 5u of the lower limit OR crit OD'd.
-.3 boost if you're crit OD'd and within 5u of the lower limit.
 All numbers NOT inclusive.

This keeps the anti-premed thing in, while keeping bonuses from higher levels. If you go over the 5u limits, you go slower. And if you take antitox to try and counter the toxic effect from OD/crit OD, you purge all of it from you.

~~What do you think would be a good penalty for "withdrawal", and how long should it last? I'm thinking 10 or so points of blurred vision, but I have to look at the normal healing rate.

Worse aim, not sure how to do it. Slowdown, I don't want to, cause it defeats the purpose as-is, but I can do blurred vision, and that seems good.~~

Regardless, I'd like to get the TM with it as-is, and see how bad it goes. I think it'll be alright, but if it isn't, or plays weirdly, then I'll change it from what it is.

Reply:
I'll do this change myself on another PR then.
For simplicity we'll change behavior  to a single speed boost on add and remove instead.

Reply:
Okay, but that just changes it to a straight-up buff.

Reply:
Actually, fuck that, I'll do the change I mentioned above. A single speed boost on add and remove is "simple", but I'd rather keep complexity and room for strategy in our gameplay. 

3 tiers, all of which require keeping within 5u of a threshold for maximum benefit, and with ascending drawbacks the higher you go both combats premed like you wanted, and keeps complexity by enabling greater benefits for greater risk.

Reply:
Change made, just need to ~~make sure it works~~ figure out how to do the movespeed add shit without it being REALLY hacky.

Reply:
Coffee changes reverted. It's both too finicky and too weak to use. I'll look at this in another PR later.